clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -137,12 +139,14 @@ public String protocolType() {
 
     @Override
     public List<ProtocolMetadata> metadata() {
+        Set<String> subscribedTopics = new HashSet<>(subscriptions.subscription());
```

Comment:
Yeah, good catch. My intent was actually to use the same object returned from `subscriptions.subscription()`, which is why I modified the code to make that object immutable after creation.